src/index.d.ts

```diff
@@ -618,22 +632,63 @@ declare module 'discord-akairo' {
 
     export type ArgumentMatch = 'phrase' | 'flag' | 'option' | 'rest' | 'separate' | 'text' | 'content' | 'restContent' | 'none';
 
-    export type ArgumentType = 'string' | 'lowercase' | 'uppercase' | 'charCodes'
-        | 'number' | 'integer' | 'bigint' | 'emojint'
-        | 'url' | 'date' | 'color'
-        | 'user' | 'users' | 'member' | 'members' | 'relevant' | 'relevants'
-        | 'channel' | 'channels' | 'textChannel' | 'textChannels' | 'voiceChannel' | 'voiceChannels' | 'categoryChannel' | 'categoryChannels' | 'newsChannel' | 'newsChannels' | 'storeChannel' | 'storeChannels'
-        | 'role' | 'roles' | 'emoji' | 'emojis' | 'guild' | 'guilds'
-        | 'message' | 'guildMessage' | 'relevantMessage' | 'invite'
-        | 'userMention' | 'memberMention' | 'channelMention' | 'roleMention' | 'emojiMention'
-        | 'commandAlias' | 'command' | 'inhibitor' | 'listener'
-        | (string | string[])[]
-        | RegExp
-        | string;
+    export interface ArgumentType {
+        string: string;
+        lowercase: string;
+        uppercase: string;
+        charCodes: string;
+        number: number;
+        integer: number;
+        bigint: bigint;
+        emojint: number;
+        url: URL;
+        date: Date;
+        color: string;
+        user: User | null;
+        users: User[] | null;
+        member: GuildMember | null;
+        members: GuildMember[] | null;
+        relevant: User | GuildMember | null;
+        relevants: User[] | GuildMember[] | null;
+        channel: Channel | null;
+        channels: Channel[] | null;
+        textChannel: TextChannel | null;
+        textChannels: TextChannel[] | null;
+        voiceChannel: VoiceChannel | null;
+        voiceChannels: VoiceChannel[] | null;
+        categoryChannel: CategoryChannel | null;
+        categoryChannels: CategoryChannel[] | null;
+        newsChannel: NewsChannel | null;
+        newsChannels: NewsChannel[] | null;
+        storeChannel: StoreChannel | null;
+        storeChannels: StoreChannel[] | null;
+        role: Role | null;
+        roles: Role[] | null;
+        emoji: Emoji | null;
+        emojis: Emoji[] | null;
+        guild: Guild | null;
+        guilds: Guild[] | null;
+        message: Message | null;
+        guildMessage: Message | null;
+        relevantMessage: Message | null;
+        invite: any;
+        userMention: User | null;
+        memberMention: GuildMember | null;
+        channelMention: GuildChannel | null;
+        roleMention: Role | null;
+        emojiMention: Emoji | null;
+        commandAlias: Command | null;
+        command: Command | null;
+        inhibitor: Inhibitor | null;
+        listener: Listener | null;
+        [key: string]: string | (string | string[]) | RegExp | any;
```

Comment:
```suggestion
        [key: string]: string | (string | string[])[] | RegExp | any;
```

Reply:
It's nicer IMO to write a complex (read: having multiple `[]`s on the same segment) type as
```diff
-[key: string]: string | (string | string[])[] | RegExp | any;
+[key: string]: string | Array<(string | string[])> | RegExp | any;
```

Reply:
Nah.

Reply:
The problem I mentioned still exists.